### PR TITLE
support vector index 

### PR DESCRIPTION
**Pull request by Jant1L**
Support create vector index and query similarity

example：
- CALL vector.AddVectorIndex('person', 'vector', 'IVF_FLAT', 4, 'L2' ,10);
- CALL vector.DeleteVectorIndex('person', 'vector', 'IVF_FLAT', 4, 'L2');
- CALL vector.ShowVectorIndex();
- CALL vector.VectorIndexQuery('person', 'vector', [1,2,3,4], 4, 5);

What it support now：
FAISS IVF_FLAT

TODO:
support more vector index algorithm and library

### CODE REVIEW

File: src/cypher/procedure/procedure.cpp [hunk 1]
```diff
@@ -3700,4 +3705,194 @@ void SpatialFunc::Distance(RTContext *ctx, const cypher::Record *record,
...
+        std::vector<int64_t> Flat_indices;
+        std::vector<std::tuple<std::string, std::string, float>> SearchResult;
+        int64_t count = 0;
+        lgraph::VectorIndex* index = extr->GetVectorIndex();
+        if (index->GetWhetherRebuild()) {
+            auto success = index->SetSearchSpec(args[4].Int());
+            auto result = index->Search(query_vector, static_cast<size_t>(args[3].Int()), index_distances, index_indices);
+            if (result && success) {
+                LOG_DEBUG() << "2";
+                auto kv_iter = schema->GetPropertyTable().GetIterator(txn.GetTxn());
```
Comment (Jant1L) [inline src/cypher/procedure/procedure.cpp:3846]: @qishipengqsp

File: src/core/lightning_graph.cpp [hunk 2]
```diff
@@ -2104,6 +2105,87 @@ bool LightningGraph::BlockingAddIndex(const std::string& label, const std::strin
...
+        schema->MarkVertexIndexed(extractor->GetFieldId(), vertex_index.release());
+        schema->MarkVectorIndexed(extractor->GetFieldId(), vector_index.release());
+
+        if ((extractor->GetVectorIndex()->GetVectorIndexManager())->MakeVectorIndex()) {
+            LOG_INFO() <<
+                FMA_FMT("set the vector index for {}:{}", label, field);
+        }
+
+        // detach property
+        if (schema->DetachProperty()) {
```
Comment (Jant1L) [inline src/core/lightning_graph.cpp:2149]: @qishipengqsp

File: docs/en-US/source/2.introduction/4.schema.md [hunk 3]
```diff
@@ -40,6 +40,7 @@ TuGraph Supports a variety of data types that can be used as attributes, the spe
 | POINT    |                     |                     | EWKB format data of point                        |
 | LINESTRING |                   |                     | EWKB format data of linestring                   |
 | POLYGON  |                     |                     | EWKB format data of polygon                      |
+| FLOAT_VECTOR |                 |                     | The dynamic vector containing 32-bit float numbers |
```
Comment (qishipengqsp) [inline docs/en-US/source/2.introduction/4.schema.md:43]: `dynamic` means?

File: include/fma-common/string_util.h [hunk 4]
```diff
@@ -672,6 +672,17 @@ inline std::string ToString(void* p) {
     return std::string(buf, r);
 }
 
+inline std::string ToString(std::vector<float> fv) {
```
Comment (qishipengqsp) [inline include/fma-common/string_util.h:675]: Has seen this before? in the last PR?

**Review by qishipengqsp — CHANGES_REQUESTED** [review]

File: src/core/vector_index_layer.h [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+ protected:
+   std::string label_;
+   std::string name_;
+   std::string distance_type_;
+   std::string index_type_;
+   int vec_dimension_;
+   std::vector<int> index_spec_;
+   int query_spec_;
+   VectorIndexManager vector_index_manager_;
```
Comment (qishipengqsp) [inline src/core/vector_index_layer.h:25]: why VectorIndex holds vectorIndexManager

**Pull request closed without merging**